standard/types.md

```diff
@@ -54,16 +54,13 @@ interface_type
     ;
 
 array_type
-    : non_array_type rank_specifier+
+    : array_type nullable_type_annotation rank_specifier+
+    | non_array_type rank_specifier+
```

Comment:
This has the property that when parsing `string?[][,]?[,,][,,,]?` we end up with two `array_type` nodes: one being `string?[][,]?[,,][,,,]`, and one being the inner `string?[][,]`.

@Nigel-Ecma It's not mutual left recursion! 😁